File: kattis/another_brick.py
# https://open.kattis.com/problems/anotherbrick
# 1s
# 2.1 Easy

# 173 chars. Top 10, still place 3
# To make it shorter, I have to rething the principal algorithm...
def f():return map(int,input().split())
h,w,_=f()
b=f()
p=0;l=0;f=1
for v in b:
 if p+v<w:p+=v
 elif p+v==w:p=0;l+=1
 else:f=0;break
r=((f==1)*(l>=h))
print(["NO","YES"][r])


"""
# Shorter: 175 chars: Top 10 Place 3
h,w,_=map(int,input().split())
b=map(int,input().split())
p=0;l=0;f=1
for v in b:
 if p+v<w:p+=v
 elif p+v==w:p=0;l+=1
 else:f=0;break
r=((f==1)*(l>=h))
print(["NO","YES"][r])
"""

"""
# Compressed: 205 chars: Top 10 Place 6
import sys;d=[list(map(int,l.split()))for l in sys.stdin.readlines()]
h,w,s=d[0];b=d[1]
p=0;l=0;f=1
for v in b:
 if p+v<w:p+=v
 elif p+v==w:p=0;l+=1
 else:f=0;break
r=((f==1)*(l>=h))
print(["NO","YES"][r])
"""


"""
# 257 chars
import sys;d=[list(map(int,l.split()))for l in sys.stdin.readlines()]
h,w,s=d[0];b=d[1]
p=0;l=0;f=1
for v in b:
    if l>=w and f==1:break
    if p+v<w:p+=v
    elif p+v==w:p=0;l+=1
    else:f=0;break
if f==1 and l>=h:
    print("YES")
else:
    print("NO")
"""

File: kattis/another_brick.pl
% https://open.kattis.com/problems/anotherbrick
% 1s
% 2.1 Easy

% 386 chars
main :-
    readln([H,W,_|B],end_of_file),
    ((s(B,H,W,0,_,0,L),L>=H)->T="YES";T="NO"),
    writeln(T).
s([],_,_,P,P,L,L).
s([B|Bs],H,W,P0,P,L0,L) :-
    (P0+B < W->
        (Bs==[] ; [T|_]=Bs, P0+B+T =< W ),
        P1 is P0+B,
        L1 = L0
    ;
        (P0+B=:=W ->
            P1 is 0,
            L1 is L0+1
        ;
            fail
        )
    ),
    s(Bs,H,W,P1,P,L1,L).

/*
% Compressed: 246 chars. Not short enough for Top 10 (98..223 chars)
main:-readln([H,W,_|B],end_of_file),((s(B,H,W,0,_,0,L),L>=H)->T="YES";T="NO"),writeln(T).
s([],_,_,P,P,L,L).
s([B|Bs],H,W,P0,P,L0,L):-(P0+B<W->(Bs==[];[T|_]=Bs,P0+B+T=<W),P1 is P0+B,L1=L0;
(P0+B=:=W->P1 is 0,L1 is L0+1;fail)),s(Bs,H,W,P1,P,L1,L).

*/

         